src/runtime/audience-action-flow.ts

```diff
@@ -321,6 +327,47 @@ export class AudienceActionFlow {
   }
 
   /**
+   * Populates localStorage with PPS configuration parameters based on
+   * SurveyDataTransferRequest.
+   **/
+  private async configureAnswerPpsData(
+    request: SurveyDataTransferRequest
+  ): Promise<void> {
+    const iabAudienceKey = StorageKeys.PPS_TAXONOMIES;
+    // PPS value field is optional and category may not be populated
+    // in accordance to IAB taxonomies.
+    const ppsConfigParams = request
+      .getSurveyQuestionsList()!
+      .flatMap((question) => question.getSurveyAnswersList())
+      .map((answer) => answer?.getPpsValue())
+      .filter((ppsValue) => ppsValue !== null);
+
+    const existingIabTaxonomy = await this.storage_.get(
+      iabAudienceKey,
+      /* useLocalStorage= */ true
+    );
+    const existingIabTaxonomyValues = existingIabTaxonomy
```

Comment:
nit: use optional chaining to avoid a runtime error when the expected key doesn't exist.

`JSON.parse(existingIabTaxonomy)?.[Constants.PPS_AUDIENCE_TAXONOMY_KEY]?.values` || []

(You don't need to nullness of `existingIabTaxonomy` as `JSON.parse(null)` evalues to `null` and it will be nicely handled by optional chaining.)

Reply:
makes sense. forgot about optional chaining since the file migrated to ts haha